aloha-core/src/main/scala/com/eharmony/aloha/dataset/vw/unlabeled/VwRowCreator.scala

```diff
@@ -145,42 +159,42 @@ extends RowCreator[A]
 
         h(nsFeatureIndices, previousNsInserted, nameAlreadyInserted = false)
     }
-}
-
-final object VwRowCreator {
 
-    private[vw] val DefaultVwNamespaceName = ""
     /**
-     * The reason to choose 17 digits is that
-     1 1 == (1 - 1.0e-17)
-     1 0.9999999999999999 == (1 - 1.0e-16).
-     * We want to retain as much information as possible without allowing long trailing sequences of zeroes.
-     */
-    private[vw] val LabelDecimalDigits = 17
-    private[vw] val LabelDecimalFormatter = new DecimalFormat(List.fill(LabelDecimalDigits)("#").mkString("0.", "", ""))
-    private[this] val labelEps = math.pow(10, -LabelDecimalDigits) / 2
-    private[this] val labelNegEps = -labelEps
-    private[vw] def labelInEpsilonInterval(label: Double) = labelNegEps < label && label < labelEps
+      *
+      * Each namespace will only be present if it contains at least one feature in the namespace.
+      * @param features features to insert into VW input line.
+      * @param defaultNamespace
+      * @param nonEmptyNamespaces
+      * @param includeZeroValues
+      * @return
+      */
+    private[aloha] def unlabeledVwInput(
+            features: IndexedSeq[Sparse],
+            defaultNamespace: List[Int],
+            nonEmptyNamespaces: List[(String, List[Int])],
+            includeZeroValues: Boolean): CharSequence = {
 
-    private[vw] val FeatureDecimalDigits = 6
-    private[vw] val DecimalFormatter = new DecimalFormat(List.fill(FeatureDecimalDigits)("#").mkString("0.", "", ""))
-    private[this] val eps = math.pow(10, -FeatureDecimalDigits) / 2
-    private[this] val negEps = -eps
-    private[vw] def inEpsilonInterval(x: Double) = negEps < x && x < eps
+        // RMD 2015-06-12: GOD I HATE THIS CODE!!!  Maybe functionalize it in the future!
 
-    final class Producer[A]
-        extends RowCreatorProducer[A, VwRowCreator[A]]
-        with RowCreatorProducerName
-        with VwCovariateProducer[A]
-        with SparseCovariateProducer
-        with CompilerFailureMessages {
+        val sb = new StringBuilder
 
-        type JsonType = VwUnlabeledJson
-        def parse(json: JsValue): Try[VwUnlabeledJson] = Try { json.convertTo[VwUnlabeledJson] }
-        def getRowCreator(semantics: CompiledSemantics[A], jsonSpec: VwUnlabeledJson): Try[VwRowCreator[A]] = {
-            val (covariates, default, nss, normalizer) = getVwData(semantics, jsonSpec)
-            val spec = covariates.map(c => new VwRowCreator(c, default, nss, normalizer))
-            spec
+        // Whether a namespace has been added previously.
+        var nsAlreadyInserted = false
```

Comment:
Is this the time to address your above TODO?

Reply:
> "Ain't nobody got time for that."